### PR TITLE
STITCH-778 - v2 login for admin

### PR DESCRIPTION
**Pull request by mburdette**
WIP still need to test this locally and with cloud

### CODE REVIEW

**Comment by mburdette** [comment]
@mpobrien updated

File: src/auth/index.js [hunk 1]
```diff
@@ -161,23 +162,25 @@ export default class Auth {
   }
 
   getAccessToken() {
-    return this.get()['accessToken'];
+    return this.get()['accessToken'] || this.get()['access_token'];
```
Comment (mburdette) [inline src/auth/index.js]: this is for client `accessToken` and admin v2 `access_token`

File: src/auth/providers.js [hunk 2]
```diff
@@ -55,6 +55,9 @@ function anonProvider(auth) {
 
 /** @namespace */
 function userPassProvider(auth) {
+  const providerRoute = auth.isClient() ? 'local/userpass' : 'providers/local-userpass';
+  const loginRoute = auth.isClient() ? 'local/userpass' : `${providerRoute}/login`;
```
Comment (mburdette) [inline src/auth/providers.js]: these login routes are the same thing as above, client v1 vs admin v2

File: src/admin.js [hunk 3]
```diff
@@ -0,0 +1,431 @@
...
+                create: (data) => api._post(`${appUrl}/auth_providers`, data),
+                authProvider: (providerId) => ({
+                  get: () => api._get(`${appUrl}/auth_providers/${providerId}`),
+                  update: (data) => api._patch(`${appUrl}/auth_providers/${providerId}`, data),
+                  enable: () => api._put(`${appUrl}/auth_providers/${providerId}/enable`),
+                  disable: () => api._put(`${appUrl}/auth_providers/${providerId}/disable`),
+                  remove: () => api._delete(`${appUrl}/auth_providers/${providerId}`)
+                })
+              }),
+              security: TODOnotImplemented,
```
Comment (UnicodeSnowman) [inline src/admin.js]: is there a ticket to implement this somewhere?

Comment (mburdette) [reply src/admin.js]: I'm not sure, this file is copy pasted from its original location in `client.js`

Comment (mpobrien) [reply src/admin.js]: Just filed https://jira.mongodb.org/browse/STITCH-836 for this

File: src/admin.js [hunk 4]
```diff
@@ -0,0 +1,431 @@
...
+  }
+
+  /**
+   * Returns profile information for the currently logged in user
+   *
+   * @returns {Promise}
+   */
+  userProfile() {
+    const api = this._v2;
+    return api._get('/auth/profile');
```
Comment (mpobrien) [inline src/admin.js]: nit: no purpose in the `const` here really. might as well just do `return this._v2._get(...)`

**Comment by mburdette** [comment]
@mpobrien there is a PR up on the ui already for the mentioned issues.

**Pull request merged**